diff -u3 -r dhcp3-3.0+3.0.1rc14/common/options.c dhcp3-3.0+3.0.1rc14-SB/common/options.c
--- dhcp3-3.0+3.0.1rc14/common/options.c	2004-06-10 17:59:19.000000000 +0000
+++ dhcp3-3.0+3.0.1rc14-SB/common/options.c	2004-06-28 15:43:47.000000000 +0000
@@ -783,6 +783,9 @@
 
 	    /* Code for next option to try to store. */
 	    code = priority_list [i];
+
+	    /* Damn M$ and their infinite stupidity. HP 2003.05.07 */
+	    if (code == 249) { continue; }
 	    
 	    /* Look up the option in the site option space if the code
 	       is above the cutoff, otherwise in the DHCP option space. */
diff -u3 -r dhcp3-3.0+3.0.1rc14/server/Makefile.dist dhcp3-3.0+3.0.1rc14-SB/server/Makefile.dist
--- dhcp3-3.0+3.0.1rc14/server/Makefile.dist	2004-06-10 17:59:50.000000000 +0000
+++ dhcp3-3.0+3.0.1rc14-SB/server/Makefile.dist	2004-06-28 15:43:47.000000000 +0000
@@ -31,7 +31,7 @@
 PROG   = dhcpd
 MAN    = dhcpd.8 dhcpd.conf.5 dhcpd.leases.5
 
-INCLUDES = -I$(TOP) $(BINDINC) -I$(TOP)/includes
+INCLUDES = -I$(TOP) $(BINDINC) -I$(TOP)/includes -I/exports/kickstart/include
 DHCPLIB = ../common/libdhcp.a $(BINDLIB) ../omapip/libomapi.a ../dst/libdst.a
 CFLAGS = $(DEBUG) $(PREDEFINES) $(INCLUDES) $(COPTS)
 
diff -u3 -r dhcp3-3.0+3.0.1rc14/server/dhcp.c dhcp3-3.0+3.0.1rc14-SB/server/dhcp.c
--- dhcp3-3.0+3.0.1rc14/server/dhcp.c	2004-06-17 20:54:40.000000000 +0000
+++ dhcp3-3.0+3.0.1rc14-SB/server/dhcp.c	2004-06-28 15:43:47.000000000 +0000
@@ -38,6 +38,9 @@
 #endif /* not lint */
 
 #include "dhcpd.h"
+/**/
+#include "kslog.h"
+/**/
 
 int outstanding_pings;
 
@@ -2695,6 +2698,10 @@
 	struct option_cache *oc;
 	char *s;
 
+	/**/
+	char *message = NULL;
+	/**/
+
 	if (!state)
 		log_fatal ("dhcp_reply was supplied lease with no state!");
 
@@ -2793,6 +2800,30 @@
 		   ? inet_ntoa (state -> giaddr)
 		   : state -> ip -> name));
 
+	/**/
+	message = make_message("dhcpd [info] %s on %s to %s via %s\n",
+		  (state -> offer
+		   ? (state -> offer == DHCPACK ? "DHCPACK" : "DHCPOFFER")
+		   : "BOOTREPLY"),
+		  piaddr (lease -> ip_addr),
+		  (lease -> hardware_addr.hlen
+		   ? print_hw_addr (lease -> hardware_addr.hbuf [0],
+				    lease -> hardware_addr.hlen - 1,
+				    &lease -> hardware_addr.hbuf [1])
+		   : print_hex_1 (lease -> uid_len, lease -> uid, 
+				  lease -> uid_len)),
+		  (state -> giaddr.s_addr
+		   ? inet_ntoa (state -> giaddr)
+		   : state -> ip -> name));
+	if (message != NULL) {
+		kslog(message);
+		free(message);
+	}
+	else {
+		kslog("dhcpd [err] Unable to allocate memory");
+	}
+	/**/
+
 	/* Set up the hardware address... */
 	hto.hlen = lease -> hardware_addr.hlen;
 	memcpy (hto.hbuf, lease -> hardware_addr.hbuf, hto.hlen);
